// gocardless-nodejs/client.js
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoCardlessClient = void 0;
const constants_1 = require("./constants");
const api_1 = require("./api/api");
const bankAuthorisationService_1 = require("./services/bankAuthorisationService");
const bankDetailsLookupService_1 = require("./services/bankDetailsLookupService");
const billingRequestService_1 = require("./services/billingRequestService");
const billingRequestFlowService_1 = require("./services/billingRequestFlowService");
const billingRequestTemplateService_1 = require("./services/billingRequestTemplateService");
const blockService_1 = require("./services/blockService");
const creditorService_1 = require("./services/creditorService");
const creditorBankAccountService_1 = require("./services/creditorBankAccountService");
const currencyExchangeRateService_1 = require("./services/currencyExchangeRateService");
const customerService_1 = require("./services/customerService");
const customerBankAccountService_1 = require("./services/customerBankAccountService");
const customerNotificationService_1 = require("./services/customerNotificationService");
const eventService_1 = require("./services/eventService");
const instalmentScheduleService_1 = require("./services/instalmentScheduleService");
const institutionService_1 = require("./services/institutionService");
const logoService_1 = require("./services/logoService");
const mandateService_1 = require("./services/mandateService");
const mandateImportService_1 = require("./services/mandateImportService");
const mandateImportEntryService_1 = require("./services/mandateImportEntryService");
const mandatePdfService_1 = require("./services/mandatePdfService");
const negativeBalanceLimitService_1 = require("./services/negativeBalanceLimitService");
const payerAuthorisationService_1 = require("./services/payerAuthorisationService");
const payerThemeService_1 = require("./services/payerThemeService");
const paymentService_1 = require("./services/paymentService");
const payoutService_1 = require("./services/payoutService");
const payoutItemService_1 = require("./services/payoutItemService");
const redirectFlowService_1 = require("./services/redirectFlowService");
const refundService_1 = require("./services/refundService");
const scenarioSimulatorService_1 = require("./services/scenarioSimulatorService");
const schemeIdentifierService_1 = require("./services/schemeIdentifierService");
const subscriptionService_1 = require("./services/subscriptionService");
const taxRateService_1 = require("./services/taxRateService");
const transferredMandateService_1 = require("./services/transferredMandateService");
const verificationDetailService_1 = require("./services/verificationDetailService");
const webhookService_1 = require("./services/webhookService");
class GoCardlessClient {
    constructor(token, environment = constants_1.Environments.Live, options = {}) {
        this._api = new api_1.Api(token, environment, options);
        this._bankAuthorisations = undefined;
        this._bankDetailsLookups = undefined;
        this._billingRequests = undefined;
        this._billingRequestFlows = undefined;
        this._billingRequestTemplates = undefined;
        this._blocks = undefined;
        this._creditors = undefined;
        this._creditorBankAccounts = undefined;
        this._currencyExchangeRates = undefined;
        this._customers = undefined;
        this._customerBankAccounts = undefined;
        this._customerNotifications = undefined;
        this._events = undefined;
        this._instalmentSchedules = undefined;
        this._institutions = undefined;
        this._logos = undefined;
        this._mandates = undefined;
        this._mandateImports = undefined;
        this._mandateImportEntries = undefined;
        this._mandatePdfs = undefined;
        this._negativeBalanceLimits = undefined;
        this._payerAuthorisations = undefined;
        this._payerThemes = undefined;
        this._payments = undefined;
        this._payouts = undefined;
        this._payoutItems = undefined;
        this._redirectFlows = undefined;
        this._refunds = undefined;
        this._scenarioSimulators = undefined;
        this._schemeIdentifiers = undefined;
        this._subscriptions = undefined;
        this._taxRates = undefined;
        this._transferredMandates = undefined;
        this._verificationDetails = undefined;
        this._webhooks = undefined;
    }
    get bankAuthorisations() {
        if (!this._bankAuthorisations) {
            this._bankAuthorisations = new bankAuthorisationService_1.BankAuthorisationService(this._api);
        }
        return this._bankAuthorisations;
    }
    get bankDetailsLookups() {
        if (!this._bankDetailsLookups) {
            this._bankDetailsLookups = new bankDetailsLookupService_1.BankDetailsLookupService(this._api);
        }
        return this._bankDetailsLookups;
    }
    get billingRequests() {
        if (!this._billingRequests) {
            this._billingRequests = new billingRequestService_1.BillingRequestService(this._api);
        }
        return this._billingRequests;
    }
    get billingRequestFlows() {
        if (!this._billingRequestFlows) {
            this._billingRequestFlows = new billingRequestFlowService_1.BillingRequestFlowService(this._api);
        }
        return this._billingRequestFlows;
    }
    get billingRequestTemplates() {
        if (!this._billingRequestTemplates) {
            this._billingRequestTemplates = new billingRequestTemplateService_1.BillingRequestTemplateService(this._api);
        }
        return this._billingRequestTemplates;
    }
    get blocks() {
        if (!this._blocks) {
            this._blocks = new blockService_1.BlockService(this._api);
        }
        return this._blocks;
    }
    get creditors() {
        if (!this._creditors) {
            this._creditors = new creditorService_1.CreditorService(this._api);
        }
        return this._creditors;
    }
    get creditorBankAccounts() {
        if (!this._creditorBankAccounts) {
            this._creditorBankAccounts = new creditorBankAccountService_1.CreditorBankAccountService(this._api);
        }
        return this._creditorBankAccounts;
    }
    get currencyExchangeRates() {
        if (!this._currencyExchangeRates) {
            this._currencyExchangeRates = new currencyExchangeRateService_1.CurrencyExchangeRateService(this._api);
        }
        return this._currencyExchangeRates;
    }
    get customers() {
        if (!this._customers) {
            this._customers = new customerService_1.CustomerService(this._api);
        }
        return this._customers;
    }
    get customerBankAccounts() {
        if (!this._customerBankAccounts) {
            this._customerBankAccounts = new customerBankAccountService_1.CustomerBankAccountService(this._api);
        }
        return this._customerBankAccounts;
    }
    get customerNotifications() {
        if (!this._customerNotifications) {
            this._customerNotifications = new customerNotificationService_1.CustomerNotificationService(this._api);
        }
        return this._customerNotifications;
    }
    get events() {
        if (!this._events) {
            this._events = new eventService_1.EventService(this._api);
        }
        return this._events;
    }
    get instalmentSchedules() {
        if (!this._instalmentSchedules) {
            this._instalmentSchedules = new instalmentScheduleService_1.InstalmentScheduleService(this._api);
        }
        return this._instalmentSchedules;
    }
    get institutions() {
        if (!this._institutions) {
            this._institutions = new institutionService_1.InstitutionService(this._api);
        }
        return this._institutions;
    }
    get logos() {
        if (!this._logos) {
            this._logos = new logoService_1.LogoService(this._api);
        }
        return this._logos;
    }
    get mandates() {
        if (!this._mandates) {
            this._mandates = new mandateService_1.MandateService(this._api);
        }
        return this._mandates;
    }
    get mandateImports() {
        if (!this._mandateImports) {
            this._mandateImports = new mandateImportService_1.MandateImportService(this._api);
        }
        return this._mandateImports;
    }
    get mandateImportEntries() {
        if (!this._mandateImportEntries) {
            this._mandateImportEntries = new mandateImportEntryService_1.MandateImportEntryService(this._api);
        }
        return this._mandateImportEntries;
    }
    get mandatePdfs() {
        if (!this._mandatePdfs) {
            this._mandatePdfs = new mandatePdfService_1.MandatePdfService(this._api);
        }
        return this._mandatePdfs;
    }
    get negativeBalanceLimits() {
        if (!this._negativeBalanceLimits) {
            this._negativeBalanceLimits = new negativeBalanceLimitService_1.NegativeBalanceLimitService(this._api);
        }
        return this._negativeBalanceLimits;
    }
    get payerAuthorisations() {
        if (!this._payerAuthorisations) {
            this._payerAuthorisations = new payerAuthorisationService_1.PayerAuthorisationService(this._api);
        }
        return this._payerAuthorisations;
    }
    get payerThemes() {
        if (!this._payerThemes) {
            this._payerThemes = new payerThemeService_1.PayerThemeService(this._api);
        }
        return this._payerThemes;
    }
    get payments() {
        if (!this._payments) {
            this._payments = new paymentService_1.PaymentService(this._api);
        }
        return this._payments;
    }
    get payouts() {
        if (!this._payouts) {
            this._payouts = new payoutService_1.PayoutService(this._api);
        }
        return this._payouts;
    }
    get payoutItems() {
        if (!this._payoutItems) {
            this._payoutItems = new payoutItemService_1.PayoutItemService(this._api);
        }
        return this._payoutItems;
    }
    get redirectFlows() {
        if (!this._redirectFlows) {
            this._redirectFlows = new redirectFlowService_1.RedirectFlowService(this._api);
        }
        return this._redirectFlows;
    }
    get refunds() {
        if (!this._refunds) {
            this._refunds = new refundService_1.RefundService(this._api);
        }
        return this._refunds;
    }
    get scenarioSimulators() {
        if (!this._scenarioSimulators) {
            this._scenarioSimulators = new scenarioSimulatorService_1.ScenarioSimulatorService(this._api);
        }
        return this._scenarioSimulators;
    }
    get schemeIdentifiers() {
        if (!this._schemeIdentifiers) {
            this._schemeIdentifiers = new schemeIdentifierService_1.SchemeIdentifierService(this._api);
        }
        return this._schemeIdentifiers;
    }
    get subscriptions() {
        if (!this._subscriptions) {
            this._subscriptions = new subscriptionService_1.SubscriptionService(this._api);
        }
        return this._subscriptions;
    }
    get taxRates() {
        if (!this._taxRates) {
            this._taxRates = new taxRateService_1.TaxRateService(this._api);
        }
        return this._taxRates;
    }
    get transferredMandates() {
        if (!this._transferredMandates) {
            this._transferredMandates = new transferredMandateService_1.TransferredMandateService(this._api);
        }
        return this._transferredMandates;
    }
    get verificationDetails() {
        if (!this._verificationDetails) {
            this._verificationDetails = new verificationDetailService_1.VerificationDetailService(this._api);
        }
        return this._verificationDetails;
    }
    get webhooks() {
        if (!this._webhooks) {
            this._webhooks = new webhookService_1.WebhookService(this._api);
        }
        return this._webhooks;
    }
}
exports.GoCardlessClient = GoCardlessClient;
